pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -260,24 +265,8 @@ func ipSecReplacePolicyOut(src, dst, tmplSrc, tmplDst *net.IPNet, dir IPSecDir)
 	return netlink.XfrmPolicyUpdate(policy)
 }
 
-func ipsecDeleteXfrmSpi(spi uint8) {
-	var err error
-	scopedLog := log.WithFields(logrus.Fields{
-		"spi": spi,
-	})
-
-	xfrmStateList, err := netlink.XfrmStateList(0)
-	if err != nil {
-		scopedLog.WithError(err).Warning("deleting previous SPI, xfrm state list error")
-		return
-	}
-	for _, s := range xfrmStateList {
-		if s.Spi != int(spi) {
-			if err := netlink.XfrmStateDel(&s); err != nil {
-				scopedLog.WithError(err).Warning("deleting old xfrm state failed")
-			}
-		}
-	}
+func ipSecPolicyForSPI(policy netlink.XfrmPolicy, spi int) bool {
+	return int(policy.Mark.Value>>12) == spi
```

Comment:
We should define this value as a constant at the top. We already use it in two places without a proper constant, but this time it's going to be easier to miss if we ever change the value. The two other places will break connectivity if they have conflicting shift values; here it may just cause XFRM states to leak, which our CI won't surface.

It can probably be a separate PR if you want to avoid another iteration through CI.